### DISCUSSION TITLE
Reading logs from a several servers

### DISCUSSION THREAD

**Discussion by andrii-bodnar** [martin-helmich/prometheus-nginxlog-exporter#270] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Hi, I'm discovering the possibility to run the exporter on a separate server and collect the Nginx logs from several servers.

I'm not sure about the `address` parameter in the `listen` block. It's the IP of the server where the exporter is running?

**Describe the solution you'd like**

A possibility to configure several servers where to collect Nginx logs.

**Describe alternatives you've considered**

Reading from syslog? Installing the exporter on each server?

**Comment by andrii-bodnar** [comment]
Hi @martin-helmich!
Could you please assist with this question?

**Comment by martin-helmich** [comment, marked as answer]
Hey Andrii; the exporter is not designed to scrape metrics from different servers. If you want to aggregate metrics from multiple servers, the generally accepted architecture for this in the Prometheus ecosystem is to have an exporter running on each server and have them scraped by a single, central Prometheus instance. Prometheus will store each of those as a separate time series (typically differentiated by a  `job` label), but you can easily aggregate using [PromQL aggregation functions](https://prometheus.io/docs/prometheus/latest/querying/functions/).

If you _really_ want, the syslog approach might work, although untested. You can configure the listen address with the `.namespaces[*].source.syslog.listen_address` property. However, you'd probably need to find some way to differentiate incoming log streams from different servers (if you don't want metrics from different servers being mushed together in a single time series).

Hope that helps!

**Reply by andrii-bodnar** [reply]
@martin-helmich thank you for the response!

**Reply by bs-amitmahajan29** [reply]
@andrii-bodnar / @martin-helmich - I had a similar use-case wherein I wanted to setup a central nginxlog exporter which will accept logs from a fleet of nginx servers. Essentially a remote syslog server which will accept data and nginxlog exporter will export that data.

I wanted to know, were you able to set it up successfully? Any challenges so far that you have encountered?

**Reply by andrii-bodnar** [reply]
Hi @bs-amitmahajan29, I have abandoned this idea and ended up with the exporter per server setup.
